tests/utils/refactoring_checks.py

```diff
@@ -0,0 +1,407 @@
+    with open(values["recipe_yaml"]) as fin:
+        recipe_hparams = load_hyperpyyaml(
+            fin, values["overrides"] | recipe_overrides
+        )
+
+    # Dataset preparation is assumed to be done through recipes; before running this.
+    exec(values["dataio"])
+    test_datasets = deepcopy(eval(values["test_datasets"]))
+
+    # harmonise
+    if type(test_datasets) is not dict:
+        tmp = {}
+        if type(test_datasets) is list:
+            for i, x in enumerate(test_datasets):
+                tmp[i] = x
+        else:
+            tmp[0] = test_datasets
+        test_datasets = tmp
+
+    # prepare testing
+    logger = FileTrainLogger(save_file=f"{tmp_dir}/{repo}.log")
+    reporting = deepcopy(values["performance"])
+    for metric, specs in reporting.items():
+        reporting[metric]["tracker"] = deepcopy(
+            recipe_hparams[specs["handler"]]()
+        )
+
+    test_loader_kwargs = deepcopy(recipe_hparams[values["test_loader"]])
+    del recipe_hparams
+
+    stats = {}
+    for k in test_datasets.keys():  # keys are test_clean, test_other etc
+        test_set = test_datasets[k]
+        if not (
+            isinstance(test_set, DataLoader)
+            or isinstance(test_set, LoopedLoader)
+        ):
+            test_set = make_dataloader(test_set, **test_loader_kwargs)
+
+        with torch.no_grad():
+            for batch in tqdm(test_set, dynamic_ncols=True, disable=False):
+                batch = batch.to(model.device)
+                wavs, wav_lens = batch.sig
+                wavs, wav_lens = (  # noqa
+                    wavs.to(model.device),
+                    wav_lens.to(model.device),
+                )
+                predictions = eval(  # noqa
+                    f'model.{values["fnx"]}(wavs, wav_lens)'
+                )
+                predicted = eval(values["predicted"])  # noqa
+                targeted = eval(values["targeted"])  # noqa
+                ids = batch.id  # noqa
+                for metric in reporting.keys():
+                    reporting[metric]["tracker"].append(*eval(values["append"]))
+
+        stats[k] = {}
+        for metric, specs in reporting.items():
+            stats[k][metric] = specs["tracker"].summarize(specs["field"])
+        logger.log_stats(
+            stats_meta=stats_meta | {"set": k}, test_stats=stats[k],
+        )
+
+    return stats
+
+
+# run first w/ "--after=False" on latest develop, then checkout the refactoring branch and run w/ "--after=True"
+# PYTHONPATH=`realpath .` python tests/integration/HuggingFace_transformers/refactoring_checks.py tests/integration/HuggingFace_transformers/overrides.yaml --LibriSpeech_data="" --CommonVoice_EN_data="" --CommonVoice_FR_data="" --IEMOCAP_data="" --after=False
+if __name__ == "__main__":
+    hparams_file, run_opts, overrides = speechbrain.parse_arguments(
+        sys.argv[1:]
+    )
+    # speechbrain.utils.distributed.ddp_init_group(run_opts)
```

Comment:
good point. I think bc I use this for single-gpu only; working on another PR that does DDP testing now. Left it in as an indicator, I think - but surely it can drop. The aim here was to test with fewest parameters possible & in debug mode.